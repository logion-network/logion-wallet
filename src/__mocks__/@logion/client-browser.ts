export class BrowserFile {

    constructor(blob: Blob) {
        this.blob = blob;
    }

    private blob: Blob;

    getBlob() {
        return this.blob;
    }
}

export class BrowserAxiosFileUploader {

    async upload() {

    }
}
