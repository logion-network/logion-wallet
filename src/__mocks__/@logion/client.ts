import { CheckResultType } from "@logion/client/dist/CollectionItem.js";
import { DefaultSignAndSendStrategy } from "@logion/client/dist/Signer.js";

import {
    LogionClient,
    AccountTokens,
    toIsoString,
    fromIsoString,
    ClosedCollectionLoc,
    DraftRequest,
    EditableRequest,
    OpenLoc,
    LocRequestState,
    LocsState,
    HashOrContent,
} from '../LogionClientMock';

export {
    LogionClient,
    AccountTokens,
    toIsoString,
    fromIsoString,
    CheckResultType,
    ClosedCollectionLoc,
    DraftRequest,
    EditableRequest,
    OpenLoc,
    DefaultSignAndSendStrategy,
    LocRequestState,
    LocsState,
    HashOrContent,
}

export { MimeType } from "@logion/client/dist/Mime.js";
