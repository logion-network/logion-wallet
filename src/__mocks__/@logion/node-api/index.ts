import { ApiPromise } from "src/__mocks__/PolkadotApiMock";
export * from "@logion/node-api/dist/UUID";

export function buildApi() {
    return new ApiPromise();
}

export * from "./dist/LogionLoc";
