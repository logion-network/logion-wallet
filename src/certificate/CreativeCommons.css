.CertificateCell .CreativeCommons .text {
    margin-left: 20px;
    margin-top: 8px;
    margin-bottom: 8px;
}
