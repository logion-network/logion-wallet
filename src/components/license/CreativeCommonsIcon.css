.CreativeCommonsIcon img {
    margin-left: 0;
}
