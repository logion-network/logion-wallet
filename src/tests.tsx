import { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import reactRenderer, { ReactTestRenderer, ReactTestRendererJSON, act as reactAct } from 'react-test-renderer';
import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

export function render(element: ReactElement): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
    const component = reactRenderer.create(element);
    return component.toJSON();
}

export function renderer(element: ReactElement): ReactTestRenderer {
    return reactRenderer.create(element);
}

export function shallowRender<E extends ReactElement>(element: ReactElement): E {
    const renderer = createRenderer();
    renderer.render(element);
    return renderer.getRenderOutput();
}

export function mockAccount(address: string, name: string) {
    return {
        address,
        meta: {
            name
        }
    };
}

export { reactAct as act };

export async function clickByName(name: string | RegExp | ((accessibleName: string, element: Element) => boolean)) {
    let button: HTMLElement;
    await waitFor(() => button = screen.getByRole("button", { name }));
    await userEvent.click(button!);
}

export async function typeByLabel(label: string, value: string) {
    let input: HTMLElement;
    await waitFor(() => input = screen.getByLabelText(label));
    await userEvent.type(input!, value);
}
