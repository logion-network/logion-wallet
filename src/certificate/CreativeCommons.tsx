import { CreativeCommons as CreativeCommonsType } from "@logion/client";
import NewTabLink from "src/common/NewTabLink";
import CertificateLabel from "./CertificateLabel";
import CreativeCommonsIcon from "../components/license/CreativeCommonsIcon";
import { Row } from "../common/Grid";
import "./CreativeCommons.css"

export interface Props {
    creativeCommons: CreativeCommonsType;
}

export default function CreativeCommons(props: Props) {

    const { creativeCommons } = props;
    return (
        <div className="CreativeCommons">
            <CertificateLabel smaller={ true }>IP rights granted with this Collection Item</CertificateLabel>
            <Row>
                <CreativeCommonsIcon creativeCommons={ creativeCommons } />
                <p className="text">This work is licensed under a&nbsp;
                    <NewTabLink href={ creativeCommons.deedUrl() } inline>
                        Creative Commons Attribution 4.0 International License
                    </NewTabLink>.
                </p>
            </Row>
        </div>
    );
}
