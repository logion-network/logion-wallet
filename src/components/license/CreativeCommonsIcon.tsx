import { CreativeCommons, CreativeCommonsCode } from "@logion/client";
import NewTabLink from "../../common/NewTabLink";
import "./CreativeCommonsIcon.css";

import by from "@creativecommons/cc-assets/license_badges/big/by.svg";
import by_sa from "@creativecommons/cc-assets/license_badges/big/by_sa.svg";
import by_nd from "@creativecommons/cc-assets/license_badges/big/by_nd.svg";
import by_nc from "@creativecommons/cc-assets/license_badges/big/by_nc.eu.svg";
import by_nc_sa from "@creativecommons/cc-assets/license_badges/big/by_nc_sa.eu.svg";
import by_nc_nd from "@creativecommons/cc-assets/license_badges/big/by_nc_nd.eu.svg";

export const creativeCommonsBadges: Record<CreativeCommonsCode, string> = {
    "BY": by,
    "BY-SA": by_sa,
    "BY-ND": by_nd,
    "BY-NC": by_nc,
    "BY-NC-SA": by_nc_sa,
    "BY-NC-ND": by_nc_nd,
}

interface CreativeCommonsIconProps {
    creativeCommons: CreativeCommons
}

export default function CreativeCommonsIcon(props: CreativeCommonsIconProps) {
    const { creativeCommons } = props;
    return (
        <NewTabLink
            className="CreativeCommonsIcon"
            href={ creativeCommons.deedUrl() }>
            <img
                src={ creativeCommonsBadges[creativeCommons.parameters] }
                alt={ creativeCommons.parameters } />
        </NewTabLink>
    )
}
