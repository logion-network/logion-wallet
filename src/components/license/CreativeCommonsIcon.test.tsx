import { shallowRender } from "../../tests";
import CreativeCommonsIcon from "./CreativeCommonsIcon";
import { CreativeCommons } from "@logion/client/dist/license/CreativeCommons";
import { UUID } from "@logion/node-api";

describe("CreativeCommonsIcon", () => {

    it("renders", () => {
        const creativeCommons = new CreativeCommons(new UUID("28e68ed4-531a-4bb5-a54e-6283bb853e82"), "BY-NC");
        const element = shallowRender(<CreativeCommonsIcon creativeCommons={ creativeCommons }/>);
        expect(element).toMatchSnapshot();
    })
})
