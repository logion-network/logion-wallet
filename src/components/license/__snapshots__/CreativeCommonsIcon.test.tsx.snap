// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreativeCommonsIcon renders 1`] = `
<NewTabLink
  className="CreativeCommonsIcon"
  href="https://creativecommons.org/licenses/by-nc/4.0/deed.en"
>
  <img
    alt="BY-NC"
    src="by_nc.eu.svg"
  />
</NewTabLink>
`;
